src/http/ngx_http_wasm_api.c

```diff
@@ -762,30 +780,56 @@ ngx_http_wasm_req_get_header(ngx_http_request_t *r, char *key,  int32_t key_size
         key_buf = (u_char *) key;
     }
 
-    for (i = 0; /* void */ ; i++) {
-
-        if (i >= part->nelts) {
-            if (part->next == NULL) {
-                break;
+    if (key_buf[0] == ':') {
+        for (i = 0; i < NGX_WASM_H2_HEADER_STATIC_TABLE_ENTRIES; i++) {
+            if (ngx_strncasecmp(key_buf, ngx_wasm_h2_header_static_table[i].name.data,
```

Comment:
We can check the len before comparing the data

Reply:
updated